### ISSUE TITLE
Files deleted after unsuccessful transcode

### ISSUE THREAD

**Issue by JayJay1974** [nwronski/batch-transcode-video#11]
After running successfully yesterday (mostly), I changed `cd` to the cwd `Videos` with two subfolders, one for my rips `BLURAY\ RIPS/` and one for the transcoded output `__Transcoded`
When I ran the batch transcode with 10 files in the rips folder as such:
`Macnesium:Videos jj$ sudo batch-transcode-video --input "BLURAY\ RIPS/" --output "__Transcoded"`
They all failed with `Error: [TranscodeError: File already exists in output directory.]` and then subsequently deleted all 10 rips.
Is this a known issue or have I done something wrong?

**Comment by nwronski** [comment]
When a destination (output) file already exists for a file from the input folder, that input file is marked as _errored_. When the batch finishes (or you terminate the program), all files marked as errored have their destination files deleted, if they exist. This is done so that you do not end up with partially-encoded or otherwise invalid video files in the output directory. If you plan to run the program using an output directory that has existing video files in it, especially files with the same file names as the input files within the input directory, you should always use the `--diff` option. When using `--diff`, input files are skipped when their corresponding output files already exist, instead of being marked as _errored_ and cleaned up (deleted).

```
batch-transcode-video --diff --input "BLURAY\ RIPS/" --output "__Transcoded"
```

**Comment by nwronski** [comment]
Since this behavior wasn't very clearly documented, and multiple people are having issues, I have added [a new section to the README](https://github.com/nwronski/batch-transcode-video#before-you-begin) to warn users. I have also published a new version ([v1.3.0](https://github.com/nwronski/batch-transcode-video/releases/tag/v1.3.0)) of the package that adds a `--keep` option to prevent `batch-transcode-video` from deleting files from the output folder, no matter what happens with the transcode process. Here is an excerpt from the README:

- `--keep` _Flag: does not accept a value_ (Default: `false`) (Alias: `-k`)
  - **Never delete any output files**, no matter what happens, **even if the encoding task fails** for the corresponding input file. If you use this option, input files that fail to encode correctly, or finish encoding, will not be deleted from the output folder. Subsequent runs, with or without using the `--diff` option, will not reprocess the failed input files, unless the corresponding output files are manually deleted.

Update to the new version with:

```
npm i -g batch-transcode-video@1.3.0
```

**Comment by JayJay1974** [comment]
Excellent. Thanks for the info and update. Much appreciated!

**Comment by JayJay1974** [comment]
Just to clarify, it deleted the _input_ files, not the output files.
Now I have to re-rip those 10 BluRays and try transcoding them.

**Comment by nwronski** [comment]
Well, that is definitely a problem. I will take another look and see what situation might cause the input file to be the one deleted instead of the output.

Not sure if it is related, but if you put a path in quotations, then you should not escape spaces with a backslash. If your directory is `BLURAY RIPS`, then you can write that as either `"BLURAY RIPS"` or `BLURAY\ RIPS`, but using `"BLURAY\ RIPS"` would point to a directory named `BLURAY\ RIPS`, based on the quick test I just did on my Mac.

**Comment by JayJay1974** [comment]
OK, I'll try that out with some test files.
I'll heed your advice as revise my input and output folder name arguments.

**Comment by nwronski** [comment]
the issue you were experiencing should be resolved in the latest release ([v2.0.0](https://github.com/nwronski/batch-transcode-video/releases/tag/v2.0.0)), but please reopen this issue if the problem persists, thanks! you can continue using the `--keep` option, but it is no longer necessary. the underlying bug, that caused your transcoded files to be deleted, has been resolved. I'd recommend still using the `--diff` option, though, so that files that have already been transcoded are skipped.

```
npm i -g batch-transcode-video@2
```
